io/io/src/TMemFile.cxx

```diff
@@ -134,13 +134,19 @@ TMemFile::TMemFile(const char *path, ExternalDataPtr_t data) :
    Init(/* create */ false);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor to create a TMemFile re-using external storage.
+
+TMemFile::TMemFile(const char *path, ExternalDataPtr_t data)
```

Comment:
This seems to significantly change the ownership semantic of this signature (taking an std::shared_ptr).  What am I missing?

Reply:
How so? This reuses code by delegating to the other non-owning constructor. The ownership remains the same and is managed by the shared_ptr if the user chose that interface.

Reply:
I see that 
```
   const ExternalDataPtr_t fExternalData; ///< shared file data / content
```
and replaced by
```
   Bool_t       fExternalData;            ///< If there is external data / content
```
I do not see where the new version can increment and later decrement the use count in the user's shared ptr.

Reply:
Ah, I see. Let me rethink that part.